packages/mui-material-next/src/InputBase/InputBase.test.tsx

```diff
@@ -595,7 +633,7 @@ describe('<InputBase />', () => {
           slots={{ input: MyInputBase }}
           slotProps={{
             input: {
-              onChange: parentHandleChange,
+              onChange: parentHandleChange as unknown as React.ChangeEventHandler<HTMLInputElement>,
```

Comment:
Could we type `parentHandleChange` as `React.ChangeEventHandler<HTMLInputElement>` instead so we don't have to cast to `unknown`?

Reply:
That was what I first tried, but the TS error message suggest casting to unknown first 😅 

```sh
/InputBase.test.tsx:636:25 - error TS2352: Conversion of type '(...args: string[]) => void' to type 'ChangeEventHandler<HTMLInputElement>' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Types of parameters 'args' and 'event' are incompatible.
    Type 'ChangeEvent<HTMLInputElement>' is not comparable to type 'string'.
```

I think `slotProps.input.onChange` expects a completely native `<input>` `onChange`, but in this test, it's called with two strings: `onChange(e.target.value, OUTPUT_VALUE)`

Reply:
I gave this test a proper look (as I was [wondering about it](https://github.com/mui/material-ui/pull/39378/files#diff-5f7200258113444091a37b1eae4beabbb8af283ef57fa74a3af6a4a09b2dd8fdR160-R162) anyway) and it was supposed to be a test for integrating `react-select` with `InputBase`, `react-select` has a custom `onChange` that is essentially `(string, string) => void`: https://github.com/mui/material-ui/issues/18130

So I've now updated this to `onChange: parentHandleChange as (string, string) => void`